### DISCUSSION TITLE
DIMENSION Text cannot be saved after modification.

### DISCUSSION THREAD

**Discussion by xbdr419** [mozman/ezdxf#984] [Q&A]
my code:

import ezdxf
f = 'Before Modification.dxf'
doc = ezdxf.readfile(f)
msp =doc.modelspace()
for e in msp:
    if e.dxftype() in ("DIMENSION", "ARC_DIMENSION", "LARGE_RADIAL_DIMENSION"):
        e.dxf.text = '6666'
doc.saveas('Modification After.dxf')
————————————————
Hello，I want to modify DIMENSION’s text，when I open  new file in autoCAD，The texts is still Before Modification。
As shown below， only  one , at bottom left corner is changed.

Modification Before 
![source](https://github.com/mozman/ezdxf/assets/19528714/56917504-9da9-4fc2-9474-5ad5463a6e7b)
Modification After
![after](https://github.com/mozman/ezdxf/assets/19528714/21d6e937-eb25-4195-aa97-ab61e5c10281)

files:
[dxf.zip](https://github.com/mozman/ezdxf/files/13578584/dxf.zip)

**Comment by mozman** [comment]
The DIMENSION entity doesn't work that way! I don't know where to start to explain the problem and I don't have the time now. I will come back in a few days to give more information about this topic.

**Comment by mozman** [comment]
The `DIMENSION` entity is very complex and requires an additional anonymous `BLOCK` that contains the rendering of the dimension as simple DXF entities like `LINE`, `MTEXT` and `INSERT` entities.

The content of that block is what a CAD application displays.

This is the reason why editing the attributes of an exiting `DIMENSION` entity is not applied immediately.

```python
import ezdxf
from ezdxf.document import Drawing

def update_block(doc: Drawing, block_name: str, value: str):
    block = doc.blocks.get(block_name)
    if block is None:
        return
    mtext = block.query("MTEXT").first
    if mtext is not None:
        mtext.dxf.text = value

def main():
    doc = ezdxf.readfile("dim.dxf")  # see attached zip-file
    msp = doc.modelspace()

    # Just modifying the text doesn't anything
    dim_linear = msp.query("DIMENSION").first
    assert dim_linear is not None
    dim_linear.dxf.text = "200"
    dim_arc = msp.query("ARC_DIMENSION").first
    assert dim_arc is not None
    dim_arc.dxf.text = "90"
    doc.saveas("dim0.dxf")

    # Edit the MTEXT content of the associated BLOCK
    update_block(doc, dim_linear.dxf.geometry, "200")
    update_block(doc, dim_arc.dxf.geometry, "90")

    doc.saveas("dim1.dxf")

if __name__ == '__main__':
    main()
```

### Original `dim0.dxf`

![dim](https://github.com/mozman/ezdxf/assets/4230966/2fad788a-f1ef-46ce-8529-735e44048011)


As soon as you change the dimension in the CAD application, the rendering will be updated and the manual set text is applied.

### `DIMENSION` entity modified by CAD application

![dim0](https://github.com/mozman/ezdxf/assets/4230966/93de66ff-999a-407a-af3a-be3fc1b2d2ab)

You can edit the `MTEXT` entity in the associated `BLOCK`. 
The name of the `BLOCK` is stored in the DXF attribute `geometry`.

### Updated `MTEXT` entity in `dim1.dxf`

![dim1](https://github.com/mozman/ezdxf/assets/4230966/d0020af9-98dc-4a70-8eb4-7b26f63f6e98)

[dim.zip](https://github.com/mozman/ezdxf/files/13618899/dim.zip)

**Reply by xbdr419** [reply]
Aha！原来如此，Thanks a million!

**Comment by xbdr419** [comment]
I hava a new questhion, I  add some text in DIMENSION .  For example ,when I add text "test" in autoCAD, the data shown 'test <>'
 
![image](https://github.com/mozman/ezdxf/assets/19528714/07b94547-dc20-48da-98c8-47156cacce56)

I try update it ,  but it can't Normal display

update_block(doc, dim_linear.dxf.geometry,  "200 <>")
update_block(doc, dim_arc.dxf.geometry, "90 <>")

![image](https://github.com/mozman/ezdxf/assets/19528714/628a21bc-da81-416a-92fc-a04c4247e4b7)

I'm not sure if there's a way to do this，let it Normal display in autoCAD
